mod_modular_vanilla/hooks/entity/tactical/actor.nut

```diff
@@ -358,10 +358,10 @@
 		{
 			local overflowDamage = _hitInfo.DamageArmor;
 
-			if (this.getBaseProperties().Armor[_hitInfo.BodyPart] != 0)
+			if ((this.getBaseProperties().Armor[_hitInfo.BodyPart] * p.ArmorMult[_hitInfo.BodyPart]) != 0)
 			{
-				overflowDamage -= this.getBaseProperties().Armor[_hitInfo.BodyPart] * this.getBaseProperties().ArmorMult[_hitInfo.BodyPart];
-				this.getBaseProperties().Armor[_hitInfo.BodyPart] = this.Math.max(0, this.getBaseProperties().Armor[_hitInfo.BodyPart] * this.getBaseProperties().ArmorMult[_hitInfo.BodyPart] - _hitInfo.DamageArmor);
+				overflowDamage -= this.getBaseProperties().Armor[_hitInfo.BodyPart] * p.ArmorMult[_hitInfo.BodyPart];
+				this.getBaseProperties().Armor[_hitInfo.BodyPart] = this.Math.maxf(0, this.getBaseProperties().Armor[_hitInfo.BodyPart] - _hitInfo.DamageArmor / p.ArmorMult[_hitInfo.BodyPart]);
```

Comment:
> Currently Normal Ifrits take 2x Armor Damage and Large Ifrits take 4x Armor Damage.
This is also the case in Vanilla.
This makes it so they currently all have only 110 effective Armor, no matter the size.

I don't understand. Could you walk me through it please? It seems to me that the damage in this function is already calculated above. This part here is only relevant for calculating `overflowDamage` which is passed to the `item_container.onDamageReceived` event. How does this in any way affect the damage received by the entity?

> After that fix, larger Ifrits will be super-buffed, but that's for another mod to deal with

Also could you please show me why will they be buffed as a result of this? Will they end up with more armor? How?